### PR TITLE
test to show noise transform fails properly

### PR DESCRIPTION
**Pull request by timmysilv**
Issue #3103 was closed automatically, but let's add one more test as per Antal's request in the closing PR.

This test demonstrates that the reported bug specifically has been fixed.

Note that if (when?) we support non-commuting observables, this will no longer raise an error and the test will fail (it will hit the `assert False` line). If we don't want this to happen, we can leave the tests as is (because the other PR more generally tests what happens when a downstream error occurs). I'm of the opinion that this is still a value-add, because we can simply remove this test when this circuit no longer raises an error!

### CODE REVIEW

**Comment by timmysilv** [comment]
won't add a new changelog entry, seems dramatic lol

File: tests/transforms/test_insert_ops.py [hunk 1]
```diff
@@ -470,3 +470,29 @@ def f2(w1, w2):
...
+    @qml.transforms.insert(noise, 0.3, position="all")
+    def noisy_circuit(circuit_param):
+        qml.RY(circuit_param, wires=0)
+        qml.Hadamard(wires=0)
+        qml.T(wires=0)
+        return qml.expval(qml.PauliX(0)), qml.expval(qml.PauliY(0)), qml.expval(qml.PauliZ(0))
+
+    try:
+        noisy_circuit(0.4)
+        assert False
```
Comment (antalszava) [inline tests/transforms/test_insert_ops.py:494]: How about using `pytest.raises` here to catch the specific error and specify its type and message?

**Review by antalszava — COMMENTED** [review]
Just one question.

**Review by antalszava — APPROVED** [review]
Looks good! :+1:

**Review by Jaybsoni — APPROVED** [review]
💯

**Pull request merged**